### ISSUE TITLE
Getting "Error: trying to read beyond partition" when configuring settings

### ISSUE THREAD

**Issue by thomasgoldstein** [raspberrypi/rpi-imager#863]
### What happened?

Hello,

TL;DR: Trying to write the Batocera image for Raspberry Pi 4 B with Raspberry Pi Imager:
* Works fine when I don't configure settings
* Fails with "Error: trying to read beyond partition" when I configure settings

![image](https://github.com/raspberrypi/rpi-imager/assets/519779/2221a920-d85f-48b0-aa6a-f87367627359)

The steps I followed to write Batocera to the card: https://wiki.batocera.org/install_batocera
Confirmed on 3 different computers and 3 different Micro SD cards.

Full context here: https://github.com/batocera-linux/batocera.linux/issues/11614

### Version

1.8.5 (Default)

### What host operating system were you using?

Windows

### Host OS Version

23H2

### Selected OS

Batocera

### Which Raspberry Pi Device are you using?

Raspberry Pi 4B, 400, and Compute Modules 4, 4S

### What kind of storage device are you using?

microSD Card in a USB reader

### OS Customisation

- [X] Yes, I was using OS Customisation when the bug occurred.

**Comment by tdewey-rpi** [comment]
Thanks for the report, @thomasgoldstein

Unfortunately you've circumvented the protections I implemented around OS customisation. Specifically, that it only be offered for operating systems we know have the infrastructure to support it.

Generally, this manifests as a "failure" to customise - the OS boots up, and the customisation doesn't apply.

In this case, however, it appears the Batocera project is forming its image in an exotic manner, and triggering errant behaviour.

As a result, I'm going to have to mark this one as "Won't Fix", and will consider implementing a stronger restriction on OS customisation.
